src/xpub/repo.rs

```diff
@@ -17,71 +28,47 @@ impl XPubs {
     }
 
     #[instrument(name = "xpubs.persist", skip(self))]
-    pub async fn persist(&self, xpub: NewAccountXPub) -> Result<XPubId, XPubError> {
-        let mut tx = self.pool.begin().await?;
-        let ret = self.persist_in_tx(&mut tx, xpub).await?;
-        tx.commit().await?;
+    pub async fn persist(&self, xpub: NewXpub) -> Result<XPubId, XpubError> {
+        let mut op = self.begin_op().await?;
+        let ret = self.persist_in_tx(&mut op, xpub).await?;
+        op.commit().await?;
```

Comment:
iiuc this made sense when the logic for persisting was hardcoded. rn its just adding additional lines with not added benefit for the whole let mut op stuff